testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java

```diff
@@ -317,4 +319,61 @@ public void testShowCreateTable()
                         Pattern.quote(getSession().getCatalog().orElseThrow()),
                         Pattern.quote(getSession().getSchema().orElseThrow())));
     }
+
+    @Test
+    public void testView()
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_VIEW)) {
+            assertQueryFails("CREATE VIEW nation_v AS SELECT * FROM nation", "This connector does not support creating views");
+            return;
+        }
+
+        String catalogName = getSession().getCatalog().orElseThrow();
+        String schemaName = getSession().getSchema().orElseThrow();
+        String viewName = "test_view_" + randomTableSuffix();
+        assertUpdate("CREATE VIEW " + viewName + " AS SELECT * FROM nation");
+
+        assertThat(query("SELECT * FROM " + viewName))
+                .skippingTypesCheck()
+                .matches("SELECT * FROM nation");
+
+        assertThat(((String) computeActual("SHOW CREATE VIEW " + viewName).getOnlyValue()))
+                .matches("(?s)" +
+                        "CREATE VIEW \\Q" + catalogName + "." + schemaName + "." + viewName + "\\E" +
+                        ".* AS\n" +
+                        "SELECT \\*\n" +
+                        "FROM\n" +
+                        "  nation");
+
+        assertUpdate("DROP  VIEW " + viewName);
+    }
+
+    @Test
+    public void testMaterializedView()
+    {
+        if (!hasBehavior(SUPPORTS_CREATE_MATERIALIZED_VIEW)) {
+            assertQueryFails("CREATE MATERIALIZED VIEW nation_mv AS SELECT * FROM nation", "This connector does not support creating materialized views");
+            return;
+        }
+
+        String catalogName = getSession().getCatalog().orElseThrow();
+        String schemaName = getSession().getSchema().orElseThrow();
+        String viewName = "test_materialized_view_" + randomTableSuffix();
+        assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " AS SELECT * FROM nation");
+        // reading
+        assertThat(query("SELECT * FROM " + viewName))
+                .skippingTypesCheck()
+                .matches("SELECT * FROM nation");
+
+        // details
+        assertThat(((String) computeActual("SHOW CREATE MATERIALIZED VIEW " + viewName).getOnlyValue()))
```

Comment:
computeScalar

Reply:
you may want to update existing tests too